larray/tests/test_array.py

```diff
@@ -918,6 +906,17 @@ def test_setitem_larray(self):
         la[:] = 0
         assert_array_equal(la, np.zeros_like(raw))
 
+        # 6) incompatible axes
+        la = self.small.copy()
+        la2 = la.copy()
+
+        with pytest.raises(ValueError):
+           la['P01'] = la2
```

Comment:
check for the error message, otherwise we have no idea whether or not the correct error was raised (and we could go back to the initial situation of not understandable message without noticing).